File: Go-Solutions/nowcoder/nc30.go
package nowcoder

func minNumberdisappered( arr []int ) int {
	// write code here
	if len(arr) == 0 {
		return 0
	}

	res := 1
	for i := 0; i < len(arr); i++ {
		if arr[i] <= 0 {
			continue
		} else if arr[i] <= res {
			res++
		}
	}
	return res
}

File: Java-Kotlin/src/main/java/nowcoder/nc30.java
package nowcoder;

import java.util.Arrays;

public class nc30 {

    public int minNumberdisappered(int[] arr) {
        // write code here
        // write code here
        if (arr.length == 0) {
            return 0;
        }
        int i = 1;           //初始化最小正整数
        Arrays.sort(arr);  //将数组排序
        for (int j = 0; j < arr.length; j++) {
            if (arr[j] <= 0) {
                continue;       //如果小于等于0则继续循环
            } else if (arr[j] <= i) {
                i++;        //如果小于等于i则i++
            }
        }
        return i;
    }

    public static void main(String[] args) {
        int[] arr = new int[]{-1, 0, 2, 3, 4};

        System.out.println(new nc30().minNumberdisappered(arr));
    }
}
